gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py

```diff
@@ -0,0 +1,424 @@
+        of generic config driver.
+
+        Returns: none
+
+        """
+
+        with mock.patch.object(
+                requests, 'post', return_value=self.resp) as mock_post, \
+            mock.patch.object(json, 'loads',
+                              return_value=self.fake_resp_dict):
+            self.driver.configure_routes(self.dict_objects.context_device,
+                                         self.kwargs)
+
+            mock_post.assert_called_with(
+                    self.dict_objects.url_for_add_src_route,
+                    data=json.dumps(self.dict_objects.data_for_add_src_route),
+                    timeout=self.dict_objects.timeout)
+
+    def test_delete_source_routes(self):
+        """ Implements test case for clear routes method
+        of generic config driver.
+
+        Returns: none
+
+        """
+
+        with mock.patch.object(
+                requests, 'delete', return_value=self.resp) as mock_delete:
+            self.driver.clear_routes(
+                self.dict_objects.context_device, self.kwargs)
+
+            mock_delete.assert_called_with(
+                    self.dict_objects.url_for_del_src_route,
+                    data=json.dumps(self.dict_objects.data_for_del_src_route),
+                    timeout=self.dict_objects.timeout)
+
+
+class VPNSvcValidatorTestCase(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(VPNSvcValidatorTestCase, self).__init__(*args, **kwargs)
+        self.dict_objects = test_vpn_data.MakeDictionaries()
+        self.plugin_rpc = vpn.VpnaasRpcSender(self.dict_objects.context,
+                                              self.dict_objects.sc)
+        self.valid_obj = vyos_vpn_driver.VPNSvcValidator(self.plugin_rpc)
+
+    def test_validate_active(self):
+        '''
+        Implements testcase for vpn driver's validate method to test in
+        success condition while making call to the service VM
+        '''
+
+        context = self.dict_objects._make_service_context()
+        svc = self.dict_objects._create_vpnservice_obj()['resource']
+        description = str(svc['description'])
+        description = description.split(';')
+        description[1] = 'tunnel_local_cidr=12.0.6.0/24'
+        description = ";".join(description)
+        svc.update({'description': description})
+
+        with mock.patch.object(self.plugin_rpc, "update_status") as mock_valid:
+            self.valid_obj.validate(context, svc)
+            mock_valid.assert_called_with(context,
+                                          self.dict_objects.vpn_vpnsvc_active)
+
+    def test_validate_error(self):
+        '''
+        Implements testcase for vpn driver's validate method to test in
+        fail condition while making call to the service VM
+        '''
+
+        context = self.dict_objects._make_service_context()
+        with mock.patch.object(self.plugin_rpc, "update_status") as mock_valid:
```

Comment:
Didn't understand how this error test case is validating the error scenario. Lets discuss.